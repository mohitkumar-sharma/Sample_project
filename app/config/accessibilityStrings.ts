export default {
  NORMAL_BUTTON_TEXT: 'normal-button-text',
  SLIDER_BUTTON_TEXT: 'slider-button-text',
  NORMAL_BUTTON: 'normal-button',
  SLIDER_BUTTON_ICON: 'slider-button-icon',
  SLIDER_BUTTON_CONTAINER: 'slider-button-container',
  BANNER_TEXT: 'banner-text',
  HEADER_SIDES_TEXT: 'header-sides-text',
  HEADER_PRESSABLE_ELEMENT: 'header-pressable-element',
  SETTINGS_SWITCH_TEXT: 'settings-switch-text',
  SETTINGS_SWITCH: 'settings-switch',
  INV_TEXT_INPUT: 'inv-text-input',
};
