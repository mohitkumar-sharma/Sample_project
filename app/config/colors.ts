export default {
  black: '#000000',
  gray: '#9B9B9B',
  divider: '#E8E8E8',
  white: '#FFFFFF',
  themeBlue: '#6EB1F7',
  themeLightBlack: '#34424A',
  appTheme: '#1A1924',
  borderLine: '#59606A',
  warning: '#FFCC00',
  red: '#FF0000',
};
