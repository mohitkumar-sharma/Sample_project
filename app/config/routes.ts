export default {
  WELCOME: 'Welcome',
  HOME: 'Home',
  SETTINGS: 'Settings',
};
