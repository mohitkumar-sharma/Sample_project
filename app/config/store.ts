import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import { rootReducer } from '../states';

const store = configureStore({
  reducer: rootReducer,
  middleware: getDefaultMiddleware(),
});

export type AppDispatch = typeof store.dispatch;

export { store };
