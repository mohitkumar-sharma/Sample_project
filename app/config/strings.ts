export default {
  GO_TO_HOME: 'Go To Home',
  DIAMOND_ICON: 'diamond',
  SLIDE_ME: 'Slide me to continue',
  APP_RUNNING_WARNING: 'Notice: You are running this app on',
  WELCOME_INPUT_PLACEHOLDER: 'Enter your name here...',
  SIGN_OUT: 'SignOut',
  HOME_BUTTONS_HEADING: '4 variations of a button',
  PRESS_ME: 'Press me',
  TRANSPARENT_BTN_PRESSED: 'Transparent button is pressed',
  LIGHT_BLACK_BTN_PRESSED: 'Light black button is pressed',
  BLUE_BTN_PRESSED: 'Blue button is pressed',
  SWITCH_TEXT: 'Enable Warning Banner',
  ALERT_TITLE_INVALID_USER_NAME: 'Invalid UserName!',
  ALERT_TITLE_BTN_PRESSED: 'Button Pressed!',
  ALERT_ENTER_VALID_USER_NAME: 'Please enter a valid user name.',
};
