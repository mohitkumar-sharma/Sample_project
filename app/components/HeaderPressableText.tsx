import React from 'react';
import { StyleSheet, StyleProp, ViewStyle, Text, TextStyle, TouchableOpacity } from 'react-native';
import config from '../config';

export interface HeaderPressableTextProps {
  /* as a string to show the pressable text for Text element*/
  text?: string;
  /* as an object of style props for TouchableOpacity element */
  containerStyle?: StyleProp<ViewStyle>;
  /* as an object of style props for Text element */
  textStyle?: StyleProp<TextStyle>;
  /* as a function triggered when pressed */
  onPress?: () => void;
}

/**
 * This is the HeaderPressableTextProps functional component for showing pressable/non-pressable text on
 * left/right side of navigation header
 * @param props : props are defined in HeaderPressableTextProps which can be passed to this component
 */
const HeaderPressableText = (props: HeaderPressableTextProps): JSX.Element => {
  const { text, containerStyle, textStyle, onPress } = props;
  return (
    <TouchableOpacity
      style={containerStyle}
      onPress={onPress}
      disabled={!onPress}
      testID={config.accessibilityStrings.HEADER_PRESSABLE_ELEMENT}>
      {text && text !== '' ? (
        <Text
          style={[styles.text, textStyle]}
          testID={config.accessibilityStrings.HEADER_SIDES_TEXT}>
          {text}
        </Text>
      ) : null}
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  text: { fontSize: 16, fontWeight: config.fontWeights.FONT_WEIGHT_REGULAR },
});

export default HeaderPressableText;
