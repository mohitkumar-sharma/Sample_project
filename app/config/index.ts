import colors from './colors';
import localConfig from './localConfig';
import strings from './strings';
import routes from './routes';
import fontWeights from './fontWeights';
import accessibilityStrings from './accessibilityStrings';
import { store } from './store';

const config = {
  colors,
  ...localConfig,
  store,
  strings,
  routes,
  fontWeights,
  accessibilityStrings,
};

export default config;
