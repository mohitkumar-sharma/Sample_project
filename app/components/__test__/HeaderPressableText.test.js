import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import HeaderPressableText from '../HeaderPressableText';
import config from '../../config';

describe('HeaderPressableText', () => {
  const props = {
    textStyle: { fontSize: 17 },
    containerStyle: { backgroundColor: config.colors.themeBlue },
  };

  const text = config.strings.SIGN_OUT;
  const textPress = jest.fn();

  it('matches snapshot', () => {
    expect(render(<HeaderPressableText {...props} />)).toMatchSnapshot();
    expect(
      render(<HeaderPressableText {...props} text={text} onPress={textPress} />)
    ).toMatchSnapshot();
  });

  it('renders the text', () => {
    const { queryByTestId } = render(<HeaderPressableText {...props} text={text} />);
    expect(queryByTestId(config.accessibilityStrings.HEADER_SIDES_TEXT)).toBeTruthy();
  });

  it('does not render the text', () => {
    const { queryByTestId } = render(<HeaderPressableText {...props} />);
    expect(queryByTestId(config.accessibilityStrings.HEADER_SIDES_TEXT)).toBeFalsy();
  });

  it('header text press', () => {
    const { getByTestId } = render(
      <HeaderPressableText {...props} text={text} onPress={textPress} />
    );

    const pressableElement = getByTestId(config.accessibilityStrings.HEADER_PRESSABLE_ELEMENT);
    fireEvent.press(pressableElement);

    expect(textPress).toHaveBeenCalledWith();
  });
});
