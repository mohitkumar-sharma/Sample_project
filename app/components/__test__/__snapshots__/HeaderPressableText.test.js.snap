// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderPressableText matches snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#6EB1F7",
      "opacity": 1,
    }
  }
  testID="header-pressable-element"
/>
`;

exports[`HeaderPressableText matches snapshot 2`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#6EB1F7",
      "opacity": 1,
    }
  }
  testID="header-pressable-element"
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 16,
          "fontWeight": "400",
        },
        Object {
          "fontSize": 17,
        },
      ]
    }
    testID="header-sides-text"
  >
    SignOut
  </Text>
</View>
`;
